policies/rudderc/src/backends/unix/ncf/method_call.rs

```diff
@@ -185,10 +186,13 @@ pub fn method_call(
     method_parameters.append(&mut specific_parameters);
     Ok((
         bundle_call,
-        Some(
-            Bundle::agent(bundle_name)
-                .parameters(method_parameters)
-                .promise_group(bundle_content),
-        ),
+        match m.resolved_foreach_state {
+            Some(ForeachResolvedState::Virtual) => None,
+            _ => Some(
```

Comment:
```suggestion
            _ => Some(
```

I'd prefer to explicitly use the other variant, as it will make adding another option safer.